backend/src/main/java/com/bakdata/conquery/apiv1/auth/ProtoUser.java

```diff
@@ -45,39 +47,28 @@
 	@NotNull
 	@Valid
 	private List<CredentialType> credentials = Collections.emptyList();
-	
-	@JsonIgnore
-	// Let this be ignored by the builder
-	private User user = null;
 
-	public User getUser() {
-		if(user != null) {
-			return user;
-		}
+	public User createOrOverwriteUser(@NonNull MetaStorage storage) {
 		if (label == null) {
 			label = name;
 		}
-		user = new User(name, label);
+		User user = new User(name, label, storage);
+		storage.updateUser(user);
+		for (String sPermission : permissions) {
+			user.addPermission(new WildcardPermission(sPermission));
+		}
 		return user;
 	}
 
-	public void registerForAuthorization(MetaStorage storage, boolean override) {
-		User user = this.getUser();
-		if(override) {			
-			storage.updateUser(user);
-		} else {
-			// Should throw an exception, if the user already existed
-			storage.addUser(user);
-		}
-		for (String sPermission : permissions) {
-			user.addPermission(storage, new WildcardPermission(sPermission));
-		}
+	@org.jetbrains.annotations.NotNull
+	public UserId getId() {
+		return new UserId(name);
 	}
-	
-	public boolean registerForAuthentication(UserManageable userManager, boolean override) {
+
+	public static boolean registerForAuthentication(UserManageable userManager, User user, List<CredentialType> credentials, boolean override) {

```

Comment:
ist das noch richtig im ProtoUser?

Reply:
Ich habe es jetzt in den AuthHelper verschoben